aws/resource_aws_kms_grant.go

```diff
@@ -0,0 +1,502 @@
+				Required: true,
+				ForceNew: true,
+			},
+			"grantee_principal": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArn,
+			},
+			"operations": {
+				Type: schema.TypeSet,
+				Set:  schema.HashString,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validateAwsKmsGrantOperation,
+				},
+				Required: true,
+				ForceNew: true,
+			},
+			"constraints": {
+				Type:     schema.TypeSet,
+				Set:      resourceKmsGrantConstraintsHash,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"encryption_context_equals": {
+							Type:     schema.TypeMap,
+							Optional: true,
+							ForceNew: true,
+							Elem:     schema.TypeString,
+							// ConflictsWith encryption_context_subset handled in Create, see kmsGrantConstraintsIsValid
+						},
+						"encryption_context_subset": {
+							Type:     schema.TypeMap,
+							Optional: true,
+							ForceNew: true,
+							Elem:     schema.TypeString,
+							// ConflictsWith encryption_context_equals handled in Create, see kmsGrantConstraintsIsValid
+						},
+					},
+				},
+			},
+			"retiring_principal": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArn,
+			},
+			"grant_creation_tokens": {
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+				ForceNew: true,
+			},
+			"grant_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"grant_token": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsKmsGrantCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).kmsconn
+
+	input := kms.CreateGrantInput{
+		GranteePrincipal: aws.String(d.Get("grantee_principal").(string)),
+		KeyId:            aws.String(d.Get("key_id").(string)),
+		Operations:       expandStringSet(d.Get("operations").(*schema.Set)),
+	}
+
+	if v, ok := d.GetOk("name"); ok {
+		input.Name = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("constraints"); ok {
+		if !kmsGrantConstraintsIsValid(v.(*schema.Set)) {
+			return fmt.Errorf("[ERROR] A grant constraint can't have both encryption_context_equals and encryption_context_subset set")
+		}
+		input.Constraints = expandKmsGrantConstraints(v.(*schema.Set))
+	}
+	if v, ok := d.GetOk("retiring_principal"); ok {
+		input.RetiringPrincipal = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("grant_creation_tokens"); ok {
+		input.GrantTokens = expandStringSet(v.(*schema.Set))
+	}
+
+	log.Printf("[DEBUG]: Adding new KMS Grant: %s", input)
+
+	var out *kms.CreateGrantOutput
+
+	err := resource.Retry(3*time.Minute, func() *resource.RetryError {
+		var err error
+
+		out, err = conn.CreateGrant(&input)
+
+		if err != nil {
+			if awsErr, ok := err.(awserr.Error); ok {
```

Comment:
You can avoid dealing with `awserr` directly by using the `isAWSErr()` helper function we have. 😄  (similarly the SDK usually provides constants for the error codes)

```go
if isAWSErr(err, kms.ErrCodeDependencyTimeoutException, "") {
```

Reply:
Done.